bundle/ScheduledVisibility/Handler/Location.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\IbexaScheduledVisibilityBundle\ScheduledVisibility\Handler;
+
+use Ibexa\Contracts\Core\Repository\ContentService;
+use Ibexa\Contracts\Core\Repository\Exceptions\BadStateException;
+use Ibexa\Contracts\Core\Repository\LocationService;
+use Ibexa\Contracts\Core\Repository\Repository;
+use Ibexa\Contracts\Core\Repository\Values\Content\Content;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Enums\HandlerType;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\ScheduledVisibility\ScheduledVisibilityInterface;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+
+use function method_exists;
+use function sprintf;
+
+final class Location implements ScheduledVisibilityInterface
+{
+    public function __construct(
+        private readonly Repository $repository,
+        private readonly ContentService $contentService,
+        private readonly LocationService $locationService,
+        private readonly LoggerInterface $logger = new NullLogger(),
+    ) {}
+
+    public function hide(Content $content): void
+    {
+        try {
+            if ($this->isHidden($content)) {
+                return;
+            }
+        } catch (BadStateException $e) {
+            $this->logger->error(
+                sprintf(
+                    'Content with id #%d is in the wrong state: %s',
+                    $content->getId(),
+                    $e->getMessage(),
+                ),
+            );
+
+            return;
+        }
+
+        if (method_exists($this->contentService, 'hideContent')) {
```

Comment:
Why is this done? According to the `composer.json` it should always be true.

Reply:
I have probably put the wrong version in composer.json but if I remember correctly(might have been dreaming) @emodric mentioned that the hideContent method is fairly new and that we should cover the case where it doesn't exist by using hideLocation from LocationService as fallback.

Reply:
It's 5 years old, not really new: https://github.com/ibexa/core/commit/f6e1114d44a023637584283588b34e6662b3475e

But let's wait for him to comment on it.

Reply:
Then by all means, we don't need it. :+1:

Reply:
fixed in 16f5f6f28154ef69948f2965fad53d9240e65d7d